update-customer/src/test/java/no/unit/nva/customer/update/UpdateCustomerHandlerTest.java

```diff
@@ -0,0 +1,130 @@
+    private UpdateCustomerHandler handler;
+    private ByteArrayOutputStream outputStream;
+    private Context context;
+
+    /**
+     * Setting up test environment.
+     */
+    @BeforeEach
+    public void setUp() {
+        customerServiceMock = mock(CustomerService.class);
+        environmentMock = mock(Environment.class);
+        when(environmentMock.readEnv(ALLOWED_ORIGIN_ENV)).thenReturn(WILDCARD);
+        handler = new UpdateCustomerHandler(customerServiceMock, environmentMock);
+        outputStream = new ByteArrayOutputStream();
+        context = new TestContext();
+    }
+
+    @Test
+    public void requestToHandlerReturnsCustomerUpdated() throws Exception {
+        UUID identifier = UUID.randomUUID();
+        Customer customer = new Customer.Builder()
+                .withIdentifier(identifier)
+                .withName("New Customer")
+                .build();
+        when(customerServiceMock.updateCustomer(identifier, customer)).thenReturn(customer);
+
+        Map<String,Object> headers = getRequestHeaders();
+        InputStream inputStream = createRequest(identifier.toString(), customer, headers);
+
+        handler.handleRequest(inputStream, outputStream, context);
+
+        GatewayResponse<Customer> actual = objectMapper.readValue(
+                outputStream.toByteArray(),
+                GatewayResponse.class);
+
+        GatewayResponse<Customer> expected = new GatewayResponse<>(
+            customer,
+            getResponseHeaders(),
+            HttpStatus.SC_OK
+        );
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void requestToHandlerWithMalformedIdentifierReturnsBadRequest() throws Exception {
+        String malformedIdentifier = "for-testing";
+        Customer customer = new Customer.Builder()
+                .withIdentifier(UUID.randomUUID())
+                .withName("New Customer")
+                .build();
+        Map<String,Object> headers = getRequestHeaders();
+        InputStream inputStream = createRequest(malformedIdentifier, customer, headers);
+
+        handler.handleRequest(inputStream, outputStream, context);
+
+        GatewayResponse<Problem> actual = objectMapper.readValue(
+                outputStream.toByteArray(),
+                GatewayResponse.class);
+
+        GatewayResponse<Problem> expected = new GatewayResponse<>(
+                Problem.builder()
+                        .withStatus(BAD_REQUEST)
+                        .withTitle(BAD_REQUEST.getReasonPhrase())
+                        .withDetail(UpdateCustomerHandler.IDENTIFIER_IS_NOT_A_VALID_UUID + malformedIdentifier)
+                        .with(REQUEST_ID, null)
+                        .build(),
+                getErrorResponseHeaders(),
+                SC_BAD_REQUEST
+        );
+
+        assertEquals(expected, actual);
+    }
+
+    protected InputStream createRequest(String identifier, Object body, Map<String,Object> headers)
```

Comment:
this is something that we need in many places during testing: 
https://github.com/BIBSYSDEV/nva-testutils/pull/3

Reply:
Good idea, but you need to extend with request parameters.